### DISCUSSION TITLE
Installation failed, cannot connect to github.com

### DISCUSSION THREAD

**Discussion by jobswalker123** [openai/whisper#1144] [Q&A] [answered]
I tired two commands from https://github.com/openai/whisper to get OpenAI-Whisper installed:

pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git 

   pip install git+https://github.com/openai/whisper.git

   and it always returns:

pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git
Collecting git+https://github.com/openai/whisper.git
  Cloning https://github.com/openai/whisper.git to c:\users\29536\appdata\local\temp\pip-req-build-_2kkh4er
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git 'C:\Users\29536\AppData\Local\Temp\pip-req-build-_2kkh4er'
  fatal: unable to access 'https://github.com/openai/whisper.git/': Recv failure: Connection was reset
  error: subprocess-exited-with-error

  × git clone --filter=blob:none --quiet https://github.com/openai/whisper.git 'C:\Users\29536\AppData\Local\Temp\pip-req-build-_2kkh4er' did not run successfully.
  │ exit code: 128
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× git clone --filter=blob:none --quiet https://github.com/openai/whisper.git 'C:\Users\29536\AppData\Local\Temp\pip-req-build-_2kkh4er' did not run successfully.
│ exit code: 128
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

But when i tried this command :

 pip install -U openai-whisper

It installed successfully.
 However, it still doesn't work
and when i tried to type "Whisper" command
it returns:

Traceback (most recent call last):
  File "C:\Users\29536\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\29536\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\29536\AppData\Local\Programs\Python\Python310\Scripts\whisper.exe\__main__.py", line 4, in <module>
  File "C:\Users\29536\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\__init__.py", line 11, in <module>
    from .audio import load_audio, log_mel_spectrogram, pad_or_trim
  File "C:\Users\29536\AppData\Local\Programs\Python\Python310\lib\site-packages\whisper\audio.py", line 8, in <module>
    import torch.nn.functional as F
ModuleNotFoundError: No module named 'torch.nn'

I tried many ways, especially for the first problems and still can't get it fixed
Any available methods to fix it?

**Comment by jongwook** [comment, marked as answer]
It seems PyTorch is not installed properly. Note that [PyTorch currently supports PyThon 3.7-3.9 on Windows](https://pytorch.org/get-started/locally/#windows-python), and you're using Python 3.10. You can follow their guides and use Chocolatey or Anaconda to install Python 3.9 and PyTorch on Windows. [PyTorch Forum](https://discuss.pytorch.org/) should also be helpful.
